.github/workflows/codespell.yml

```diff
@@ -0,0 +1,28 @@
+name: Codespell test
+
+on:
+  push:
+    branches: [ 'master', 'main', 'release/**' ]
+  pull_request:
+    branches: [ '*' ]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+# END OF COMMON SECTION
+
+jobs:
+  codespell:
+    if: github.repository_owner == 'wolfssl'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+
+    - uses: codespell-project/actions-codespell@v2.1
+      with:
+        check_filenames: true
+        check_hidden: true
+          # Add comma separated list of words to ignore (sorted alphabetically, case sensitive)
+        ignore_words_list: adin,ADn,alph,aNULL,carryIn,cconfiguration,chainG,ciph,cLen,cliKs,creen,dout,ede,Fo,haveA,hashIn,larg,LEAPYEAR,inCreated,inOut,inout,Merget,mot,optionA,parm,parms,repid,rIn,rsource,ser,siz,Te,te,TOOM,userA,vie
+        exclude_file: ./IDE/ECLIPSE/DEOS/tls_wolfssl.c,./IDE/ECLIPSE/MICRIUM/client_wolfssl.c,./IDE/IAR-MSP430/main.c,./tests/api.c
```

Comment:
A comment about why we are excluding certain files would be nice.

Reply:
Added comments.